### ISSUE TITLE
Allow authentication mechanism clientcert to provide role as well

### ISSUE THREAD

**Issue by mikn** [floragunncom/search-guard#418]
We are running ES with SG in a highly dynamic environment (kubernetes), where we cannot predict CN for all client certificates (think log shippers and dashboards), but we would still like to be able to trace audit logs to a single instance.
We would like to be able to assign group through the `clientcert` provider, by designating one of the DN fields for backend role, such as O or OU.

**Comment by floragunncom** [comment]
With "group" i assume you mean "role"?

Can you provide an example or maybe even a PR?

**Comment by mikn** [comment]
Correct, I mean role. :)
An example would be, as for the proxy_auth mechanism,
```yaml
clientcert_auth_domain:
  http_enabled: true
  transport_enabled: true
  order: 1
  http_authenticator:
    type: clientcert
    config:
      username_attribute: cn
      roles_attribute: ou  # role binding, ou can be defined multiple times
    challenge: false
    authentication_backend:
      type: noop
```
Will submit a PR on something I cobbled together.

**Comment by floragunncom** [comment]
Will be released with the next 5.6.x and 6.x release

**Comment by floragunncom** [comment]
Released with SG v17 (For 5.6.x)
